src/backend/parser/parse_relation.c

```diff
@@ -1453,10 +1453,15 @@ parserOpenTable(ParseState *pstate, const RangeVar *relation, int lockmode)
 	if (rel == NULL)
 	{
 		if (relation->schemaname)
+		{
+			const char * schema_name = relation->schemaname;
+			if (sql_dialect == SQL_DIALECT_TSQL && remove_db_name_in_schema_hook)
+				schema_name = (*remove_db_name_in_schema_hook)(schema_name);
 			ereport(ERROR,
 					(errcode(ERRCODE_UNDEFINED_TABLE),
 					 errmsg("relation \"%s.%s\" does not exist",
-							relation->schemaname, relation->relname)));
+							schema_name, relation->relname)));
+		}
```

Comment:
Here I think it is better to have a separate if stmt with erreport,  it is not recommended to to directly modify PG code.
You can do something like this: 
```
if (TSQL and hook function is defined)
  call hook function (within hook function, you raise the error)
else
 original ereport.
```
Also for the schema_name, can you simply use relation->schemaname?